dev/com.ibm.ws.cdi.internal/src/com/ibm/ws/cdi/internal/interfaces/ExtensionArchive.java

```diff
@@ -43,4 +43,7 @@ public interface ExtensionArchive extends CDIArchive {
      * @return a set of suppliers of extension objects
      */
     Set<Supplier<Extension>> getSPIExtensionSuppliers();
+
+    default void addLiteExtensionTranslator(Extension liteExtensionTranslator, Set<String> buildCompatibleExtensionClassNames) {};
```

Comment:
This method is weird and takes a completely different approach to everything else on this interface. Every other option is set via the constructor, why would this one have a setter?

Reply:
Initially, I thought there was only one implementation of this interface so I was confused as to why we have two: the general purpose `ExtensionArchiveImpl` and the special case `ProbeExtensionArchive`.

This method would only work on `ExtensionArchiveImpl`, so I'm becoming more and more convinced it shouldn't be on this interface.

Reply:
Furthermore, it doesn't strictly need to be a Lite Extension Translator extension, right? You're just adding a list of class names to be reported as "extension classes" and an extension instance you'd like to be added to the extension archive.

Reply:
You're correct that it doesn't strictly need to be a Lite Extension Translator. 

We could call it something like addConstructedExtension instead. As this method exists because we need to pass in Lite Extension Translator objects rather than just a class name.

Reply:
It uses a setter simply because I thought it would be less ugly to call a setter from cdi 4.0 ExtensionArchiveProvider than to have the other three pass in nulls. 

If you'd prefer to use the constructor that would be an easy change.

Reply:
I think `addConstructedExtension` is a good name.

As discussed, I think this does need to come off the interface because we have three implementations (I forgot `LiteExtensionArchive`) and only one will ever implement it.

I don't mind whether we move this to the constructor (which is a bit unwieldy because it has two arguments) or cast to the implementation to call the method.